content/en/blog/_posts/Hardware-Accelerated-SSLTLS-Termination-in-Ingress-Controllers-using-Kubernetes-Device-Plugins-and-RuntimeClass.md

```diff
@@ -0,0 +1,168 @@
+upon Pod creation.
+
+## Device Assignment and Pod Deployment
+
+At this point, you may ask what the container could do with a VFIO device node? The answer comes
+after we first take a quick look into the Kubernetes RuntimeClass.
+
+The Kubernetes RuntimeClass was created to provide better control and configurability
+over a variety of *runtimes* (an earlier [blog post](https://kubernetes.io/blog/2018/10/10/kubernetes-v1.12-introducing-runtimeclass/) goes into the details of the needs,
+status and roadmap for it) that are available in the cluster. In essence, the RuntimeClass
+provides cluster users better tools to pick and use the runtime that best suits for the pod use case.
+
+The OCI compatible [Kata Containers runtime](https://katacontainers.io/) provides workloads with a hardware virtualized
+isolation layer. In addition to workload isolation, the Kata Containers VM has the added
+side benefit that the VFIO devices, as `Allocate`’d by the device plugin, can be passed
+through to the container as hardware isolated devices. The only requirement is that the
+Kata Containers kernel has driver for the exposed device enabled.
+
+That’s all it really takes to enable hardware accelerated crypto for container workloads. To summarize:
+
+  1. Cluster needs a device plugin running on the node that provides the hardware,
+  2. Device plugin exposes the hardware to user space using  the VFIO driver,
+  3. Pod requests the device resources and Kata Containers as the RuntimeClass in the PodSpec,
+  4. The container has the hardware adaptation library and the OpenSSL engine module
+
+Figure 2 shows the overall setup using the Container A illustrated earlier.
+
+{{<figure width="600"  src="/images/blog/2019-XX-YY-foo/k8s-blog-fig2.png" caption="Figure 2. Deployment overview">}}
+
+## Reference Setup
+
+Finally, we describe the necessary building blocks and steps to build a functional
+setup described in Figure 2 that enables hardware accelerated SSL termination in
+an Ingress Controller using an Intel&reg; QuickAssist Technology (QAT) PCIe device.
+It should be noted that the use cases are not limited to Ingress controllers, but
+any OpenSSL based workload can be accelerated.
+
+### Cluster configuration:
+  * Kubernetes 1.14 (`RuntimeClass` and `DevicePlugin` feature gates enabled (both are `true` in 1.14)
+  * RuntimeClass ready runtime and Kata Containers configured
+
+### Host configuration:
+  * Intel&reg; QAT driver release with the kernel drivers installed for both host kernel and Kata Containers kernel (or on a rootfs as loadable modules)
+  * [QAT device plugin](https://github.com/intel/intel-device-plugins-for-kubernetes/tree/master/cmd/qat_plugin) DaemonSet deployed
+
+### Ingress controller configuration and deployment:
+  * [HAproxy-ingress](https://github.com/jcmoraisjr/haproxy-ingress)* ingress controller in a modified container that has
+     * the QAT HW HAL user space library (part of Intel&reg; QAT SW release) and
```

Comment:
Do you need the asterisk after `https://github.com/jcmoraisjr/haproxy-ingress)*`

Reply:
The asterisk is a footnote. I've now replaced it with ```[^1]```

Reply:
```[^1]``` did not render like I wanted so I've just dropped it completely.